azurerm/data_source_container_orchestrator_versions.go

```diff
@@ -0,0 +1,78 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmKubernetesServiceVersions() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmKubernetesServiceVersionsRead,
+
+		Schema: map[string]*schema.Schema{
+			"location": locationSchema(),
+			"version_prefix": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"versions": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+			"latest_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceArmKubernetesServiceVersionsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).containerServicesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	location := d.Get("location").(string)
+
+	listResp, err := client.ListOrchestrators(ctx, location, "managedClusters")
+	if err != nil {
+		if utils.ResponseWasNotFound(listResp.Response) {
+			return fmt.Errorf("Error: No Kubernetes Service versions found for location %q", location)
+		}
+		return fmt.Errorf("Error retrieving Kubernetes Versions in %q: %+v", location, err)
+	}
+
+	lv, err := version.NewVersion("0.0.0")
+	if err != nil {
+		return fmt.Errorf("Cannot set version baseline (likely an issue in go-version): %+v", err)
+	}
+	versions := []string{}
+	for _, rawV := range *listResp.Orchestrators {
+		if rawV.OrchestratorType != nil && rawV.OrchestratorVersion != nil {
+			if *rawV.OrchestratorType == "Kubernetes" {
```

Comment:
given the history with some of the API's, we probably want to be doing this case insensitively:

```suggestion
			if strings.EqualsFold(*rawV.OrchestratorType, "Kubernetes") {
```